components/blitz/src/omero/gateway/model/ROIData.java

```diff
@@ -247,7 +252,13 @@ public int getShapeCount()
      */
     public List<ShapeData> getShapes(int z, int t)
     {
-        return roiShapes.get(new ROICoordinate(z, t));
+        List<ShapeData> res = roiShapes.get(new ROICoordinate(z, t));
+        if (res == null)
+            res = new ArrayList<ShapeData>();
+        List<ShapeData> allZT = roiShapes.get(new ROICoordinate(-1, -1));
+        if (allZT != null)
+            res.addAll(allZT);
+        return res;
```

Comment:
What about if only one of *z* or *t* is -1?

Reply:
Argh, that's true, these cases have to be considered, too.

Reply:
:+1: Longer-term for the API might be nice to say something like: `new ROICoordinate(new Optional(z), new Optional(t))` or `new LenientROICoordinate(z, t)` or `new ROICoordinate(z, t, lenient=true)`